### PR TITLE
[internal/splunk] Splunk metric names should be validated via regex

### PR DESCRIPTION
**Pull request by spiffyy99**
**Description:** For the splunk HEC receiver, add additional criteria for the metric names.  As per https://docs.splunk.com/Documentation/Splunk/9.2.1/Metrics/Overview#What_is_a_metric_data_point.3F, metric names can only consist of specific characters. Validates these requirements to be in line with other Splunk receivers.

**Link to tracking Issue:** [#34275](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34275)

**Testing:** Unit tests

**Documentation:** N/A

### CODE REVIEW

File: internal/splunk/common.go [hunk 1]
```diff
@@ -63,8 +81,11 @@ func (e *Event) GetMetricValues() map[string]any {
 
 	values := map[string]any{}
 	for k, v := range e.Fields {
-		if strings.HasPrefix(k, "metric_name:") {
-			values[k[12:]] = v
+		// only consider metric name if it fits regex criteria.
+		// use matches[1] since first element contains entire string.
+		// first subgroup will be the actual metric name.
```
Comment (atoulme) [inline internal/splunk/common.go:86]: should this comment be in the `getMetricNameFromField` function?

Comment (spiffyy99) [reply internal/splunk/common.go:86]: will move

Comment (spiffyy99) [reply internal/splunk/common.go]: moved

**Review by atoulme — APPROVED** [review]

File: .chloggen/splunk_enforce_metrics_naming_regex.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: internal/splunk
```
Comment (dmitryax) [inline .chloggen/splunk_enforce_metrics_naming_regex.yaml:7]: This is only applicable to the hec receiver, right? Let's make it more user-friendly

```suggestion
component: receiver/splunkhec
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**